protocol/standard-l2-genesis.md

```diff
@@ -0,0 +1,128 @@
+the summary should be as succinct as possible. -->
+
+The L2 predeploys are refactored in a way such that the network specific configuration
+is all sourced from a single location where it is ultimately set from L1 deposit transactions.
+Any initializable logic in the L2 predeploys is also removed, to make the deposit transaction
+based upgrade scheme simple to reason about.
+
+This will accelerate the ability to ship secure software, as we will be able to get chains
+on the same versions of the software and know which versions work very well together.
+
+# Problem Statement + Context
+
+<!-- Describe the specific problem that the document is seeking to address as well
+as information needed to understand the problem and design space.
+If more information is needed on the costs of the problem,
+this is a good place to that information. -->
+
+There is currently no good way to do releases of L2 predeploys. Historically, chains
+have launched with an arbitrary commit for their L2 genesis, making the block history integrity
+checks that are part of the superchain registry very difficult. We tell chains that are
+trying to launch to use governance approved L1 contracts, the same should apply for L2.
+
+Given all of the work for making releases and upgrades nice for the L1 contracts and the client software,
+it is at a waste if we cannot also have good releases of the L2 predeploys.
+
+Right now, OP Mainnet is running contracts at various versions of the software. It is actually very difficult
+to reproduce the exact OP Mainnet set of contracts being used. It would require cherry picking bytecode from many
+different commits. We run no tests against this particular combination of contracts. We believe it is safe
+given our development practices, but every time that we do want to do a release it results in a lot of time
+being spent trying to make sure that we are upgrading to a compatible set of contracts.
+
+# Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+A WIP implementation can be found [here](https://github.com/ethereum-optimism/optimism/pull/12057).
+The specs can be found [here](https://github.com/ethereum-optimism/specs/tree/17ef36cdc3bb9893b206a93464122d56730d30fb/specs/protocol/holocene).
+
+Similar to the L1 MCP project, we move all of the network specific configuration out of the individual contracts
+themselves and instead place all of it in a single place. The contracts will make a `CALL` rather than using
+`sload` to read the values. These values will be sourced from L1 via deposit transactions that come from
+the `SystemConfig.initialize` call. We need to make sure that the max deposit gas limit is at least able to
+fullfill these deposit transactions.
```

Comment:
Do we also need to increase `l1InfoGasOverhead`?

https://github.com/ethereum-optimism/op-geth/blob/d5a96613c22bc46238a21d6c0f805399c26c9d4c/core/txpool/validation.go#L37

Reply:
@tynes why do we need to benchmark?

I think @mds1 is correct here: 

> ... we do not since these are dedicated `setConfig` calls

ie. we are not calling `setL1BlockValues()` or `setL1BlockValuesEcotone()`, and are simply emitting a `TransactionDeposited()` event, so the upgrade call is only limited by the [`maxResourceLimit`](https://github.com/ethereum-optimism/optimism/blob/feat/holocene-contracts/packages/contracts-bedrock/src/L1/ResourceMetering.sol#L36).

Reply:
Its not safe to not meter any deposit tx until we have deposit queue since it could be possible to create a block that uses more than the block gas limit. Technically you could not meter up to a certain point but what that point is is related to the chain config which is dynamic and potentially arbitrary up to certain points

Reply:
I am confident that these changes do not impact on the `l1InfoGasOverhead` although it does need to be taken into account as it determines the remaining 'buffer' which has been discussed elsewhere. That buffer discussion is tracked here: https://github.com/ethereum-optimism/optimism/issues/12578